tensorflow_addons/image/interpolate_spline_test.py

```diff
@@ -0,0 +1,365 @@
+
+                # Compute scipy interpolation.
+                scipy_interp_function = sc_interpolate.interp1d(
+                    train_points_, train_values_, kind='linear')
+
+                scipy_interpolation = scipy_interp_function(query_points_)
+                scipy_interpolation_on_train = scipy_interp_function(
+                    train_points_)
+
+                # Even with float64 precision, the interpolants disagree with scipy a
+                # bit due to the fact that we add the EPSILON to prevent sqrt(0), etc.
+                tol = 1e-3
+
+                self.assertAllClose(
+                    train_values_,
+                    scipy_interpolation_on_train,
+                    atol=tol,
+                    rtol=tol)
+                self.assertAllClose(
+                    interp_, scipy_interpolation, atol=tol, rtol=tol)
+
+    def test_1d_interpolation(self):
+        """Regression test for interpolation with 1-D points."""
+
+        tp = _QuadraticPlusSinProblem1D()
+        (query_points, _, train_points,
+         train_values) = tp.get_problem(dtype='float64')
+
+        for order in (1, 2, 3):
+            for reg_weight in (0, 0.01):
+                interpolator = interpolate_spline(train_points, train_values,
+                                                  query_points, order,
+                                                  reg_weight)
+
+                target_interpolation = tp.HARDCODED_QUERY_VALUES[(order,
+                                                                  reg_weight)]
+                target_interpolation = np.array(target_interpolation)
+                with self.cached_session() as sess:
+                    interp_val = sess.run(interpolator)
+                    self.assertAllClose(interp_val[0, :, 0],
+                                        target_interpolation)
+
+    def test_nd_linear_interpolation(self):
+        """Regression test for interpolation with N-D points."""
+
+        tp = _QuadraticPlusSinProblemND()
+        (query_points, _, train_points,
+         train_values) = tp.get_problem(dtype='float64')
+
+        for order in (1, 2, 3):
+            for reg_weight in (0, 0.01):
+                interpolator = interpolate_spline(train_points, train_values,
+                                                  query_points, order,
+                                                  reg_weight)
+
+                target_interpolation = tp.HARDCODED_QUERY_VALUES[(order,
+                                                                  reg_weight)]
+                target_interpolation = np.array(target_interpolation)
+                with self.cached_session() as sess:
+                    interp_val = sess.run(interpolator)
+                    self.assertAllClose(interp_val[0, :, 0],
+                                        target_interpolation)
+
+    def test_nd_linear_interpolation_unspecified_shape(self):
```

Comment:
To use placeholder in the test case, we could test it with decorator `test_utils.run_deprecated_v1`.
Another way to test the shape of None at graph build time is https://github.com/tensorflow/addons/issues/260#issuecomment-494886404.